### ISSUE TITLE
Losses accept a target whose shape merely broadcasts against the input and return a wrong loss

### ISSUE THREAD

**Issue by vyncint** [vyncint/oxmera#48]
## Summary

None of the three losses validates the target's shape. They rely on binary broadcasting, so a target that is broadcastable but wrong produces a finite, plausible, wrong loss.

## Where (main at 853a224)

- `crates/oxmera-nn/src/loss.rs:16` — `let diff = input.sub(target)?;` (MSE)
- `crates/oxmera-nn/src/loss.rs:38` — `one_hot(&target …, classes)` with no check on `target.dims()` (CrossEntropy checks only `logits.ndim() == 2`, line 30)
- `crates/oxmera-nn/src/loss.rs:58` — `let xz = logits.mul(target)?;` (BCEWithLogits)

## Reproduction

```rust
CrossEntropyLoss.forward(&logits_4x3, &target_of_shape_1)?;   // Ok: one label applied to four rows
BCEWithLogitsLoss.forward(&Tensor::zeros([4, 1]), &Tensor::ones([4]))?; // Ok: averaged over a [4, 4] broadcast
MSELoss.forward(&Tensor::zeros([4, 1]), &Tensor::ones([4]))?;           // Ok: same [4, 4] broadcast
```

PyTorch raises on all three.

## Why it matters

A label tensor mis-shaped by a stray `squeeze`/`unsqueeze` is one of the most common training bugs. Here it trains a model on the wrong objective with no signal. The house rule is that a precondition failure is a typed error naming the input and the rule.

## Fix

- `CrossEntropyLoss`: require `target.dims() == [n]` and `DType::I64`; error otherwise.
- `MSELoss`, `BCEWithLogitsLoss`: require `target.shape() == input.shape()`; return `Error::ShapeMismatch { op: "MSELoss", expected, got }`.
- One test per loss per wrong shape.

## Done when

The three reproductions above return `Err` with a message naming the loss, the expected shape and the shape given.
